src/adapters/rubicon.js

```diff
@@ -77,6 +83,72 @@ function RubiconAdapter() {
     });
   }
 
+  function buildVideoRequestPayload(bid) {
+    bid.startTime = new Date().getTime();
+
+    let oParams = bid.params;
+
+    if(!oParams || typeof oParams.inventory !== "object") {
+      console.log("Invalid Video Bid - params not correctly defined");
+      throw "Invalid Video Bid";
+    }
+
+    let postData =  {
+      page_url: utils.getTopWindowUrl(),
+      resolution:  window.screen.width +'x'+ window.screen.height,
+      account_id: oParams.accountId,
+      integration: 'pbjs.lite',
+      timeout: 1000,
+      stash_creatives: true,
+      slots: [],
+    };
+
+    // Define the slot object
+    let slotData = {
+      site_id: oParams.siteId,
+      zone_id: oParams.zoneId,
+      position: oParams.position || 'btf',
+      floor: 0.01,
+      element_id: oParams.elemId,
+      name: oParams.name
+    };
+
+    // make sure the custom video object is defined correctly
+    if(oParams.video && typeof oParams.video === "object") {
+      slotData.language = oParams.video.language;
+      slotData.height = oParams.video.playerHeight;
+      slotData.width = oParams.video.playerWidth;
+      postData.ae_pass_through_parameters = oParams.video.aeParams;
+    } else {
+      console.log("Invalid Video Bid - video object");
+      throw "Invalid Video Bid!";
+    }
+
+    // check and add inventory, keywords, visitor and size_id data
+    if(Array.isArray(oParams.sizes) && oParams.sizes.length > 0) {
+      slotData.size_id = oParams.sizes[0];
```

Comment:
video size id does not have a mapping - it is a misnomer - it actually refers to the type of video ad, i.e. pre-roll, mid-roll etc. so the value is required to execute a bid.